File: couch-server/src/main/java/com/fronde/server/services/validation/rule/DateFutureRule.java
package com.fronde.server.services.validation.rule;

import com.fronde.server.services.validation.ValidationResultFactory;
import java.util.Date;

public class DateFutureRule extends AbstractRule {

  @Override
  public void validate(Object object, Object propertyValue, ValidationResultFactory resultFactory) {
    if (propertyValue != null) {
      if (propertyValue instanceof Date) {
        Date dateValue = (Date) propertyValue;
        if (!dateValue.after(new Date())) {
          resultFactory.addResult("DateFuture");
        }
      } else {
        throw new RuntimeException(
            "This rule class is expecting a Date but was given " + propertyValue.getClass()
                .getName());
      }
    }
  }
}

File: angular-client/src/app/form/validators/date-future-rule.ts
import { AbstractControl, ValidationErrors } from '@angular/forms';
import * as moment from 'moment';
import { AbstractRule } from './abstract-rule';
import { ValidationResultFactory } from './validation-result-factory';

export class DateFutureRule extends AbstractRule {
  validate(control: AbstractControl): ValidationErrors | null {
    const resultFactory = new ValidationResultFactory(
      'DateFutureRule',
      this.validationService
    );

    if (control.value) {
      if (!moment(control.value).isAfter(moment())) {
        resultFactory.addResult('DateFuture');
      }
    }

    return resultFactory.validationErrors;
  }
}
